lib/dash/segment_template.js

```diff
@@ -465,16 +465,18 @@ shaka.dash.SegmentTemplate = class {
           info.unscaledPresentationTimeOffset;
       const repId = context.representation.id;
       const bandwidth = context.bandwidth || null;
+      const mediaTemplate = info.mediaTemplate;
```

Comment:
Thank you so much for the fix!  But without context, it's hard to see why the code has changed.  It would be very very easy for someone to refactor and create a regression here.

Can you please add comments explaining the rules to be followed here to avoid a leak?

I'm inferring this from your code and the PR message, so it might be wrong, but here's an example of the sort of thing I would like to see:

```js
  // This callback must not capture any non-local
  // variables, such as info, context, etc.  Make
  // sure any values you reference here have
  // been assigned to local variables within the
  // loop, or else we will end up with a leak.
  const createUris = () => {
```

Reply:
Sure.